src/test/java/com/snowflake/kafka/connector/internal/SinkServiceIT.java

```diff
@@ -239,24 +243,28 @@ public void testRecordNumber() throws Exception
         .addTask(table, topic, partition1)
         .build();
 
-    Future<Integer> result = insert(service, partition, numOfRecord);
-    Future<Integer> result1 = insert(service, partition1, numOfRecord1);
+    insert(service, partition, numOfRecord);
+    insert(service, partition1, numOfRecord1);
 
-    assert result.get() == numOfRecord / numLimit;
-    assert result1.get() == numOfRecord1 / numLimit;
+    TestUtils.assertWithRetry(
+        () -> getStageSize(stage, table, partition) == numOfRecord / numLimit, 5, 4);
+    TestUtils.assertWithRetry(
+        () -> getStageSize(stage, table, partition1) == numOfRecord1 / numLimit, 5, 4);
 
-    TestUtils.assertWithRetry(() ->
-      TestUtils.tableSize(table) == numOfRecord + numOfRecord1, 30, 6);
+    TestUtils.assertWithRetry(() -> {
+        service.insert(new ArrayList<>()); // trigger time based flush
+        return TestUtils.tableSize(table) == numOfRecord + numOfRecord1;
+      }, 30, 8);
 
     service.closeAll();
   }
 
-  private Future<Integer> insert(SnowflakeSinkService sink, int partition,
```

Comment:
this insert function does not check file number on stage anymore, because flush is async now.